url8417b1b213_Ratchette_Rummikub_tgz-pJ8-rummikub_RummikubJ8/src/rummikub/RAIclient.java

```diff
@@ -0,0 +1,272 @@
+			round = 0;
+			
+		} catch (Exception e) {
+			System.out.println("[FATAL ERROR] Could not connect to the server at " + serverIP);
+			System.out.println("Please ensure that you have started the server before the client");
+			
+			e.printStackTrace();
+			System.exit(0);
+		}
+	}
+	
+	// **********************************************************
+    // 				Main + User input validation
+    // **********************************************************
+
+	/**
+	 * Performs user input validation, then runs the AI client as a thread
+	 * @param args command line arguments sent to main
+	 */
+	public static void main(String[] args){
+		String serverIP;
+		int port;
+		
+		serverIP = Server.SERVER_IP;
+		port = Server.PORT_NUM;
+		
+		if(args.length > 0)
+			serverIP = args[0];
+		if(args.length > 1)
+			port = Integer.parseInt(args[1]);
+		if(args.length > 2){
+			System.out.println("Invalid command line options: " + args.toString());
+			System.out.println("\nProper syntax of this program is ");
+			System.out.println("\tRAIclient.java <Server IP> <port number>");
+			System.out.println("\tServer IP defaults to " + Server.SERVER_IP);
+			System.out.println("\tport number defaults to " + Server.PORT_NUM);
+			System.exit(-1);
+		}
+		
+		(new RAIclient(serverIP, port)).start();
+	}
+	
+	/**
+	 * The "main" function of the client thread.
+	 */
+	@Override
+	public void run() {
+		ArrayList<Meld> play, trivialPlay;
+		Boolean playMade;
+		
+		try {
+			startNewGame();
+			
+			while (!game.isGameOver()) {
+				updateRound();
+				
+				playMade = false;
+				
+				System.out.println(game.displayGame());
+				printStatus("Current Hand : " + hand.toString());
+				
+				if(initialMeld){
+					trivialPlay = hand.getMeldsFromHand();
+					
+					if(trivialPlay.size() > 0){
+						game.addMelds(trivialPlay);
+						game.setHand(playerIndex, hand.getNumTiles());
+						playMade = true;
+					}
+					
+					play = game.getAdjacentPlay(hand, playerIndex);
+					if(play != null){
+						playMade = true;
+					}
+					else if (trivialPlay.size() > 0)
+						play = trivialPlay;
+				}
+				
+				else{
+					play = hand.getInitialMeld();
+					
+					if(play != null){
+						game.addMelds(play);
+						game.setHand(playerIndex, hand.getNumTiles());
+						initialMeld = true;
+						playMade = true;
+					}
+				}
+				
+				if(!playMade){
+					printStatus("Could not make a meld");
+					drawTile();
+				}
+				else{
+					playMelds(play);
+				}
+				
+				// Wait for it to be your turn again 
+				game = new GameInfo(client.getMessage());
+			}
+		}
+
+		catch (Exception e) {
+			printStatus("The server disconnected me?");
+			e.printStackTrace();
+		}
+		
+		this.endGame();
+	}
+	
+	/**
+	 * Prints the current status of the client
+	 * 
+	 * @param message
+	 *            What is going on in the client
+	 */
+	private void printStatus(String message) {
+		String date = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss").format(new Date());
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The SimpleDateFormat object was created without setting the time zone. Make sure that you want to use the default time zone or use letter z, Z or X in the pattern. Otherwise, if you do not call setTimeZone() on the created SimpleDateFormat object, you might get an unexpected date and time when using the object.